File: go/main/test/all_longest_strings_test.go
package main

import (
	"reflect"
	"testing"
)

func TestAllLongestStrings(t *testing.T) {
	data := []string{"aba", "aa", "ad", "vcd", "aba"}
	got := AllLongestStrings(data)
	want := []string{"aba", "vcd", "aba"}
	if !reflect.DeepEqual(got, want) {
		t.Errorf("AllLongestStrings(#{data}) = #{got} want #{want}")
	}
}

File: php/tests/AllLongestStringsTest.php
<?php

use PHPUnit\Framework\TestCase;
require "src/all_longest_strings.php";

final class AllLongestStringsTest extends TestCase
{
    public function testAllLongestStrings()
    {
        $data = ["aba", "aa", "ad", "vcd", "aba"];
        $actual = all_longest_strings($data);
        $expected = ["aba", "vcd", "aba"];
        $this->assertEquals($expected, $actual);
  }
}
